event-hubs.api.md

```diff
@@ -0,0 +1,303 @@
+}
+
+// @public
+export interface GetPropertiesOptions extends AbortSignalOptions, SpanOptions {
+}
+
+// @public
+export class InMemoryPartitionManager implements OwnershipManager, CheckpointManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    // (undocumented)
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export interface LastEnqueuedEventInfo {
+    enqueuedTime?: Date;
+    offset?: string;
+    retrievalTime?: Date;
+    sequenceNumber?: number;
+}
+
+export { MessagingError }
+
+// @public
+export interface OwnershipManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+}
+
+// @public
+export interface PartitionCheckpointer {
+    updateCheckpoint(eventData: ReceivedEventData): Promise<void>;
+    updateCheckpoint(sequenceNumber: number, offset: number): Promise<void>;
+    // @internal (undocumented)
+    updateCheckpoint(eventDataOrSequenceNumber: ReceivedEventData | number, offset?: number): Promise<void>;
+}
+
+// @public
+export interface PartitionContext {
+    consumerGroupName: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    partitionId: string;
+}
+
+// @public
+export type PartitionManager = CheckpointManager & OwnershipManager;
+
+// @public
+export interface PartitionOwnership extends PartitionContext {
+    eTag?: string;
+    lastModifiedTimeInMS?: number;
+    offset?: number;
+    ownerId: string;
+    ownerLevel: number;
+    sequenceNumber?: number;
+}
+
+// @public
+export interface PartitionProperties {
+    beginningSequenceNumber: number;
+    eventHubName: string;
+    lastEnqueuedOffset: number;
+    lastEnqueuedSequenceNumber: number;
+    lastEnqueuedTimeUtc: Date;
+    partitionId: string;
+}
+
+// @public
+export type ProcessCloseHandler = (reason: CloseReason, context: PartitionContext & PartitionCheckpointer) => Promise<void>;
+
+// @public
+export type ProcessErrorHandler = (error: Error, context: PartitionContext) => Promise<void>;
+
+// @public
+export type ProcessEvents = (receivedEvent: ReceivedEventData, context: PartitionContext & PartitionCheckpointer) => Promise<void>;
+
+// @public
+export type ProcessInitializeHandler = (context: PartitionContext) => Promise<void>;
+
+// @public
+export interface ReceivedEventData {
+    body: any;
+    enqueuedTimeUtc: Date;
+    offset: number;
+    partitionKey: string | null;
+    properties?: {
+        [key: string]: any;
+    };
+    sequenceNumber: number;
+    systemProperties?: {
```

Comment:
system properties does not have a type. Its a bag of key value pair

Reply:
@bterlson was wondering if there are any known properties that we should call out as actual fields/properties rather than just offering up the bag only.

Reply:
There are 3 known system properties that are already being pulled up to be first class citizens of `ReceivedEventData`. These are
- offset
- sequence number
- enqueued time

When the "track last enqueued event info" boolean is enabled, the below are available on the system properties, and we do the needful to pull them up as well
- offset of the last enqueued event
- sequence number of the last enqueued event
- enqueued time of the last enqueued event

Apart from these, when you are using the Event Hubs library to receive events from IotHub, the system properties contain a few more "properties". 

Currently, the expectation is to keep the system properties as it is i.e as a dictionary/key-value bag

Reply:
Seems like we're okay with the bag then @bterlson  - all the "known" properties have already been promoted to be first class citizens.